demo/src/lib/gtag.js

```diff
@@ -0,0 +1,34 @@
+const GA_TRACKING_ID = process.env.REACT_APP_GA_TRACKING_ID;
+
+/**
+ * Sends a page_view event to Google Analytics.
+ * @param {string} pagePath
+ */
+export const pageview = (pagePath) => {
+  window.gtag('config', GA_TRACKING_ID, {
+    page_path: pagePath,
+  });
+};
+
+/**
+ * Sends an event to Google Analytics.
+ * Can be spoofed to return mock code snippet of event instead.
+ * View the [Developer Guide](https://developers.google.com/analytics/devguides/collection/app-web/events)
+ * to review recommended events and their associated parameters.
+ * @param {string} event
+ * @param {!Object<string, *>} parameters
+ * @param {boolean=} spoof If true, returns mock code instead of sending event.
+ * @return {?string} Code snippet of the event being sent.
```

Comment:
Not particularly wrong but the provide boolean essentially splits this method into two. It does look like a violation of single responsibility principle and it's a bit hard to reason about what this the main purpose of this method and at what circumstances the return is expected.

I'm not sure if supporting such spoofing is even necessary but in case it is
consider having two separate methods or always returning the code snippet (so caller doesn't have to null-check the return value).

Reply:
Yeah this was debated a bit amongst ourselves. We don't want it to be returned every time because we will need the code snippet before firing off the event in a lot of cases, but we also were unsure of splitting the function up because there are [several functions built on top of this one which we would also need to split up](https://github.com/googleinterns/measurement-library/pull/25/files#diff-2bfb813f8773f5d91020c916ca5ae49d).

I do agree it violates the SRP and can indeed split this and the other functions up if you recommend it, though!

Reply:
To add a bit more context: We are trying to build code snippets based on the state of our site in the same way we build events based off of the site state.

Reply:
As an update: I went ahead and separated out the two functions since it is indeed two different things we are trying to do :)